lib/jsdom/living/helpers/dates-and-times.js

```diff
@@ -106,7 +106,7 @@ function serializeYearlessDate({ month, day }) {
   return `${monthStr}-${dayStr}`;
 }
 
-const timeRe = /^([0-9]{2}):([0-9]{2})(?::([0-9]{2}(?:\.[0-9]{1,3})?))?$/;
+const timeRe = /^([0-9]{2}):([0-9]{2})(?::([0-9]{2}(?:\.([0-9]{1,3}))?))?$/;
```

Comment:
it's better to parse milliseconds directly in order to avoid loss of accuracy due to truncation and subtraction, eg. 1.001 - Math.trun(1.001) =/= 0.001, this lead to a bunch of failures with range overflow/underflows as the millis weren't being parsed correctly.